source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,278 @@
+``openUploadStream()`` method allows you to specify configuration information
+such as file chunk size and other field/value pairs to store as metadata. Set
+these options as parameters of ``openUploadStream()`` as shown in the
+following code snippet:
+
+.. code-block:: javascript
+
+   fs.createReadStream('./myFile').
+        pipe(bucket.openUploadStream('myFile', {
+            chunkSizeBytes: 1048576,
+            metadata: { field: 'myField', value: 'myValue' }
+        });
+
+See the :node-api-4.0:`openUploadStream() API documentation <classes/gridfsbucket.html#openuploadstream>` for more information.
+
+.. _gridfs-retrieve-file-info:
+
+Retrieve File Information
+-------------------------
+
+In this section, you can learn how to retrieve file metadata stored in the
+``files`` collection of the GridFS bucket. The metadata contains information
+about the file it refers to, including:
+
+- The id of the file
+- The name of the file
+- The length/size of the file
+- The upload date and time
+- A ``metadata`` document in which you can store any other information
+
+To retrieve files from a GridFS bucket, call the ``find()`` method on
+the ``GridFSBucket`` instance. The method returns a ``FindCursor`` instance
+from which you can access the results.
+
+The following code example shows you how to retrieve and print file metadata
+from all your files in a GridFS bucket. Among the different ways that you can
+traverse the retrieved results from the ``FindCursor`` iterable, the example
+below uses the ``forEach()`` method to display the results:
+
+.. code-block:: javascript
+
+   const cursor = bucket.find({});
+   cursor.forEach(doc => console.log(doc));
+
+The ``find()`` method accepts various query specifications and can be
+combined with other methods such as ``sort()``, ``limit()``, and ``project()``.
+
+For more information on the classes and methods mentioned in this section,
+see the following resources:
+
+- :node-api-4.0:`find() API documentation <classes/gridfsbucket.html#find>`
+- :node-api-4.0:`FindCursor API documentation <classes/findcursor.html>`
+- :doc:`Cursor Fundamentals page </fundamentals/crud/read-operations/cursor>`
+- :doc:`Read Operations page </fundamentals/crud/read-operations/>`
+
+.. _gridfs-download-files:
+
+Download Files
+--------------
+
+You can download files from your MongoDB database using a download
+stream. ``GridFSBucket`` contains the ``openDownloadStreamByName()``
+method which creates a download stream. 
+
+The example below shows you how to download a file referenced
+by the file name, stored in the ``filename`` field, into your working
+directory. Alternatively, you can use the ``openDownloadStream()``
+method, which takes the ``_id`` field of a file as a parameter. Note
+that if there are multiple documents with the same ``filename`` value,
+GridFS will stream the most recent file with the given name (as
+determined by the ``uploadDate`` field).
```

Comment:
Suggestion: I think you should show the snippet right after saying what the snippet will show. Then after it, you can continue with the last two sentences. It might also be beneficial in showing how this alternative would look like.

```suggestion
directory. Alternatively, you can use the ``openDownloadStream()``
method, which takes the ``_id`` field of a file as a parameter. 
If there are multiple documents with the same ``filename`` value,
GridFS will stream the most recent file with the given name (as
determined by the ``uploadDate`` field).
```

Reply:
accepted

Reply:
Although it may be obvious in how to do the alternative, can you provide a snippet just so the user has a visual on how they can implement it?

Reply:
included an example